libkbfs/tlf_journal.go

```diff
@@ -183,6 +183,16 @@ type tlfJournal struct {
 	mdJournal    *mdJournal
 	disabled     bool
 	lastFlushErr error
+	// The cache of unflushed paths.  If `unflushedReady` is not nil,
+	// then any callers must wait for it to be closed before accessing
+	// `unflushedPaths`.  `unflushedReady` only transitions
+	// nil->non-nil->nil at most one time during the lifetime of the
+	// journal. If `unflushedPaths` is non-nil, then `chainsPopulator`
+	// must also be non-nil.  These three fields are protected by
+	// `journalLock`.
+	unflushedPaths  map[MetadataRevision]map[string]bool
+	unflushedReady  chan struct{}
+	chainsPopulator chainsPathPopulator
```

Comment:
I thought about passing this in at construction time, but journals are enabled via `libfs/journal_control_file.go`, and they don't have access to a `folderBlockOps` at that point.  Do you think we should change it so that enabling the journal goes through `folderBranchOps`?